extensions/artillerytables/artillerytables.cpp

```diff
@@ -0,0 +1,356 @@
+}
+#endif
+
+// Constants
+static const double timeStep = 1.0 / 100;
+static const double rangeSearchErrorMax = 0.001; // ratio * distance
+static const double rangeSearchAngleConvergance = 0.00001;
+static const double gravityABS = 9.8066;
+static const ace::vector3<double> gravityAccl(0, 0, -1 * gravityABS);
+
+// Globals:
+std::vector<std::future<std::string>> fWorkers;
+unsigned int getLineIndex = 0;
+
+std::tuple<double, double, double> simulateShot(const double& _fireAngleRad, const double& _muzzleVelocity, const double& _heightOfTarget, const double& _crossWind, const double& _tailWind, const double& _temperature, const double& _airDensity, double _airFriction) {
+    const double kCoefficient = -1.0 * _airDensity * _airFriction;
+    const double powderEffects = (_airFriction) ? ((_temperature + 273.13) / 288.13 - 1) / 40 + 1 : 1.0;
+
+    double currentTime = 0;
+    ace::vector3<double> currentPosition(0, 0, 0);
+    ace::vector3<double> lastPosition(currentPosition);
+    ace::vector3<double> currentVelocity(0, powderEffects * _muzzleVelocity * cos(_fireAngleRad), powderEffects * _muzzleVelocity * sin(_fireAngleRad));
+    const ace::vector3<double> wind(_crossWind, _tailWind, 0);
+
+    while ((currentVelocity.z() > 0) || (currentPosition.z() >= _heightOfTarget)) {
+        lastPosition = currentPosition;
+        ace::vector3<double> apparentWind = wind - currentVelocity;
+        ace::vector3<double> changeInVelocity = gravityAccl + apparentWind * (kCoefficient * apparentWind.magnitude());
+        currentVelocity += changeInVelocity * timeStep;
+        currentPosition += currentVelocity * timeStep;
+        currentTime += timeStep;
+    }
+
+    const double lastCurrentRatio((_heightOfTarget - currentPosition.z()) / (lastPosition.z() - currentPosition.z()));
+    ace::vector3<double> finalPos = lastPosition.lerp(currentPosition, lastCurrentRatio);
+
+    return { finalPos.x(), finalPos.y(), currentTime };
+}
+
+std::tuple<double, double> findMaxAngle(const double& _muzzleVelocity, const double& _airFriction) {
+    if (_airFriction == 0) {
+        return { M_PI_4, _muzzleVelocity * _muzzleVelocity / gravityABS };
+    }
+    double bestAngle = M_PI_4;
+    double bestDistance = -1;
+    double testResultDist;
+    for (double testAngle = M_PI_4; testAngle > 0; testAngle -= (M_PI_4 / 100.0)) {
+        std::tie(std::ignore, testResultDist, std::ignore) = simulateShot(testAngle, _muzzleVelocity, 0, 0, 0, 15, 1, _airFriction);
+        if (testResultDist > bestDistance) {
+            bestAngle = testAngle;
+            bestDistance = testResultDist;
+        }
+    }
+    return { bestAngle, bestDistance };
+}
+
+std::tuple<double, double, double> simulateFindSolution(const double& _rangeToHit, const double& _heightToHit, const double& _muzzleVelocity, const double& _airFriction, const double& _minElev, const double& _maxElev, const bool& _highArc) {
```

Comment:
`const bool&` is less effective than just bool. Need to transmit a full 32/64 bit memory address instead of just a single byte.

Reply:
Well in theory & should only be used for classes/structs. In that case, types like double or int should just be passed by value... but there is nothing set on stone except maybe on MISRA ruleset

Reply:
Yeah I thought the same. But technically on 32bit you still get benefit if you pass a 64bit double by ref because it's bigger than the pointer and if you don't have xmm registers available to hold it.
But not really worth it.